src/data/matrix/basic.lean

```diff
@@ -57,6 +57,11 @@ ext_iff.mp
 
 end ext
 
+/-- Cast a function into a matrix -/
```

Comment:
The reason for this definition is not immediately obvious, so let's explain it here. (Maybe also a note on the definition of `matrix` itself?)
```suggestion
/-- Cast a function into a matrix.

The two sides of the equivalence are actually equal types. We want to use an explicit cast
to distinguish the types because `matrix` has different instances than pi types (such as `has_mul`).
-/
```